src/overlays/actors/ovl_Bg_Dy_Yoseizo/z_bg_dy_yoseizo.c

```diff
@@ -38,56 +37,723 @@ ActorInit Bg_Dy_Yoseizo_InitVars = {
+                    if (play->msgCtx.currentTextId == 0x59A) {
+                        Message_CloseTextbox(play);
+                    }
+                    this->timer = 20;
+                    return;
+                }
+            }
+
+            if (play->msgCtx.currentTextId != 0x59A) {
+                // "Hold B and then release"
+                Message_StartTextbox(play, 0x59A, &this->actor);
+            }
+        }
+    }
+
+    if (csAction != this->csAction) {
+        switch (csAction) {
+            case 9:
+                Animation_PlayLoop(&this->skelAnime, sAnimations[GREATFAIRY_ANIM_ARMS_FOLDED]);
+                break;
+
+            case 10:
+                Animation_PlayLoop(&this->skelAnime, sAnimations[GREATFAIRY_ANIM_CLAPPING]);
+                break;
+
+            case 11:
+                Animation_PlayOnce(&this->skelAnime, sAnimations[GREATFAIRY_ANIM_TEACH_SPIN_ATTACK]);
+                break;
+        }
+
+        this->csAction = csAction;
+    }
+}
+
+// Choose behaviour?
+void func_80A0BB08(BgDyYoseizo* this, PlayState* play) {
+    if (Cutscene_IsCueInChannel(play, CS_CMD_ACTOR_CUE_103) &&
+        (play->csCtx.actorCues[Cutscene_GetCueChannel(play, CS_CMD_ACTOR_CUE_103)]->id == 2)) {
+        func_80A0B834(this);
+        this->actionFunc = func_80A0B75C;
+    }
+
+    if (Cutscene_IsCueInChannel(play, CS_CMD_ACTOR_CUE_103) &&
+        (play->csCtx.actorCues[Cutscene_GetCueChannel(play, CS_CMD_ACTOR_CUE_103)]->id == 7)) {
+        this->actor.draw = BgDyYoseizo_Draw;
+        Animation_PlayLoop(&this->skelAnime, sAnimations[GREATFAIRY_ANIM_LAYING_DOWN]);
+        this->actionFunc = func_80A0B184;
+        this->mouthIndex = 0;
+        this->actor.world.pos.y = this->actor.home.pos.y + 40.0f;
+        Actor_SetScale(&this->actor, 0.035f);
+        this->unk2F8 = 0;
+    }
+
+    if (Cutscene_IsCueInChannel(play, CS_CMD_ACTOR_CUE_103) &&
+        (play->csCtx.actorCues[Cutscene_GetCueChannel(play, CS_CMD_ACTOR_CUE_103)]->id == 9)) {
+        Actor_SetScale(&this->actor, 0.01f);
+        Animation_PlayLoop(&this->skelAnime, sAnimations[GREATFAIRY_ANIM_ARMS_FOLDED]);
+        this->csAction = 9;
+        this->actionFunc = BgDyYoseizo_TrainPlayer;
+        this->actor.draw = BgDyYoseizo_Draw;
+    }
+}
+
+void BgDyYoseizo_Update(Actor* thisx, PlayState* play) {
+    BgDyYoseizo* this = THIS;
+
+    this->actionFunc(this, play);
+    Actor_MoveWithGravity(&this->actor);
+
+    DECR(this->timer);
+
+    BgDyYoseizo_UpdateEffects(this, play);
+}
+
+s32 BgDyYoseizo_OverrideLimbDraw(PlayState* play, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, Actor* thisx) {
+    BgDyYoseizo* this = THIS;
+
+    if (limbIndex == GREAT_FAIRY_LIMB_TORSO) {
+        rot->x += this->torsoRot.y;
+    }
+
+    if (limbIndex == GREAT_FAIRY_LIMB_HEAD) {
+        rot->x += this->headRot.y;
+        rot->z += this->headRot.z;
+    }
+
+    return false;
+}
+
+// Colour and shape of eyebrows, hair colour.
```

Comment:
C-style commenting `/** */` vs `//` not sure what the consensus is on function commenting. 
```suggestion
// Colour and shape of eyebrows, hair colour.
```
```suggestion
// Colour and shape of eyebrows, hair colour.
```